Neos.Neos/Tests/Behavior/Features/Fusion/ContentCache.feature

```diff
@@ -0,0 +1,66 @@
+@flowEntities @contentrepository
+Feature: Tests for Fusion ContentCache
+  Background:
+    Given I have Fusion content cache enabled
+    And I have the following Fusion setup:
+    """fusion
+    include: resource://Neos.Fusion/Private/Fusion/Root.fusion
+    include: resource://Neos.Neos/Private/Fusion/Root.fusion
+
+    prototype(Neos.Neos:Test.ContentCache) < prototype(Neos.Fusion:Component) {
+        foo = ''
+        renderer = ${props.foo}
+        @cache {
+            mode = 'cached'
+            entryIdentifier {
+                test = 'test'
+            }
+        }
+    }
+
+"""
+
+  Scenario: Render a cached prototype and check if rerendering doesn't happen on second try
+    When I execute the following Fusion code:
+    """fusion
+        test = Neos.Neos:Test.ContentCache {
+            foo = 'some-cached-string'
+        }
+    """
+    Then I expect the following Fusion rendering result:
+    """
+    some-cached-string
+    """
+    When I execute the following Fusion code:
+    """fusion
+        test = Neos.Neos:Test.ContentCache {
+            foo = 'some-other-string'
+        }
+    """
+    Then I expect the following Fusion rendering result:
+    """
+    some-cached-string
+    """
```

Comment:
Hmm i understand the idea but this is not how fusion works normally ... it. doesnt change its fusion code.

I rather have the test follow how fusion works:

1. have one big setup
2. trigger different entrypoints or have different outside word variables

or do you think differently?

Reply:
Yes, maybe. But actually the test work as expected 🤔 And actually this test it's just ensuring, that a caching is enabled.

Reply:
Hmm, but if I think about this... 

The tests I need this for, will look the same. I need somehow check, if a already cached entry got changed, because of a cache invalidation.

Reply:
Imo what we want to test is this:


```
    Given I have Fusion content cache enabled
     And I have the following Fusion setup:
     """fusion
     include: resource://Neos.Fusion/Private/Fusion/Root.fusion
     test = Neos.Fusion:Value {
         value = ${myVariable}
         @cache {
             mode = 'cached'
             entryIdentifier {
                 test = 'test'
             }
         }
     }
     """
     When i render the path 'test' with the following context:
     """json
     {myVariable:'initial'}
     """
     I Expect the output 'initial'
     When i render the path 'test' with the following context:
     """json
     {myVariable:'other'}
     """
     I Expect the output 'initial'
```

that way the test does exactly what we do when we run fusion in multiple requests.
Changing the fusion configuration at runtime might work in this test environment but is not how fusion is supposed to work and this behaviour might actually be buggy. So id prefer that ^^

Reply:
I like and IMO we could have a test for both scenarios

Reply:
A very realistic use case would be to render things based on a node and then change that node instance (just looking at the from my mobile atm, maybe you already do this)

Reply:
Ah, IIRC this is exactly the preparation for such tests to cover the behaviour of the content cache flusher, right?

Reply:
I'd like to keep it as it is. We use this existing feature already in multiple test cases and all I want to add with this PR is the ability to enable tjhe content cache. I guess an change in "how to test fusion" would be nice, but is currently not on my todo list.

Reply:
@dlubitz for the record: my blabbering above was meant as "this _could_ be done as follow up", I already approved this one.
However, Marc-Henry has a point ofc. We should try to make those tests as realistic as possible.
@mhsdesign what do you think of merging this one as is and then tweaking this (and the other!) features along the lines you suggested?

Reply:
Yes, yes... I know. I was just trying to convince @mhsdesign to postpone that 😅

Reply:
Okay i see im overruled thats fine ^^. And i also agree that its unfair to block denny while i approved the initial pr that introduced this syntax https://github.com/neos/neos-development-collection/pull/4641 i was thinking at that time that this might be a good idea :) So its definitely no drama and we would gain a lot if we have tests regarding the content cache in any syntax so thanks for dealing with this ❤️